src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -160,7 +159,7 @@ public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
                 }
 
                 if (!bundleConfigured
-                    || extensionItem.Bindings.Count == 0
+                    || extensionItem.Bindings is null || extensionItem.Bindings.Count == 0
```

Comment:
With the new changes, `Bindings` property will be `null` when it is not present in the extension JSON content. (see the new test added in the PR). In that case, previously, the Bindings property was populated, but as an empty collection. With STJ, It will not be populated as empty collection, but as null when the source JSON does not have it.